### DISCUSSION TITLE
What happened xmlns with "primefaces" or the URI of primefaces?

### DISCUSSION THREAD

**Discussion by ChenYouRen** [primefaces/community#2800] [PrimeFaces] [answered]
Excuse me, I don't quite understand xmlns...
In discussion-2750 example it was written `xmlns:p="primefaces"`; it can work.
But it doesn't work, when I wrote it in my code.
It only work when I write `xmlns:p="http://primefaces.org/ui"`; otherwise it's a blank page and it have not any error.

**Comment by melloware** [comment]
`xmlns:p="http://primefaces.org/ui"` - JSF 2.3 or lower

`xmlns:p="primefaces"` = Faces 4.0 or higher

**Reply by ChenYouRen** [reply]
Excuse me, I still have a problem:
I use `FacesContext.class.getPackage().getImplementationVersion()` to check my JSF version. It's 4.0.7.
And my pom.xml setting is:
```
<dependency>
	<groupId>jakarta.platform</groupId>
	<artifactId>jakarta.jakartaee-api</artifactId>
	<version>10.0.0</version>
	<scope>provided</scope>
</dependency>
<dependency>
    <groupId>org.primefaces</groupId>
    <artifactId>primefaces</artifactId>
    <version>14.0.0</version>
    <classifier>jakarta</classifier>
</dependency>
```
Why I can't use `xmlns:p="primefaces"`?

**Reply by melloware** [reply, marked as answer]
you need 

```xml
<dependency>
    <groupId>org.primefaces</groupId>
    <artifactId>primefaces</artifactId>
    <version>14.0.5</version>
    <classifier>jakarta</classifier>
</dependency>
```

**Reply by ChenYouRen** [reply]
Embarrassed..., thanks a lot.
